src/analyzing/AST_to_IL.ml

```diff
@@ -1384,30 +1453,8 @@ and stmt_aux env st =
       (* HACK(new): Because of field-sensitivity hacks, we need to know to which
        * variable are we assigning the `new` object, so we intercept the assignment. *)
       let obj' = var_of_name obj in
-      let obj_lval = lval_of_base (Var obj') in
-      let ss1, args' = args_with_pre_stmts env (Tok.unbracket args) in
-      let opt_cons =
-        let* cons = mk_class_constructor_name ty cons_id_info in
-        let cons' = var_of_name cons in
-        let cons_exp =
-          mk_e
-            (Fetch
-               {
-                 obj_lval with
-                 rev_offset = [ { o = Dot cons'; oorig = NoOrig } ];
-               })
-            (SameAs (G.N cons |> G.e))
-          (* THINK: ^^^^^ We need to construct a `SameAs` eorig here because Pro
-           * looks at the eorig, but maybe it shouldn't? *)
-        in
-        Some cons_exp
-      in
-      let ss2, ty = type_with_pre_stmts env ty in
-      ss1 @ ss2
-      @ [
-          mk_s
-            (Instr (mk_i (New (obj_lval, ty, opt_cons, args')) (SameAs new_exp)));
-        ]
+      let lval = lval_of_base (Var obj') in
+      mk_class_construction env lval new_exp ty cons_id_info args
```

Comment:
Pass the name `obj'` to `mk_class_construction` rather than `lval`, that code is assuming that we have a `x = new T()` kind of statement, it's not correct if the LHS is not a simple variable so we shouldn't be passing it a (potentially) arbitrary l-value..